pkg/controller/servicemesh/memberroll/controller.go

```diff
@@ -323,6 +342,19 @@ func (r *MemberRollReconciler) reconcileObject(ctx context.Context, roll *maistr
 		kialiErr = r.kialiReconciler.reconcileKiali(ctx, kialiName, meshNamespace, allKnownMembers.List())
 	}
 
+	// 7. tell Prometheus about all the namespaces in the mesh
+	var prometheusErr error
+
+	if mesh.Status.AppliedSpec.IsPrometheusEnabled() {
+		cv, err := versions.ParseVersion(mesh.Status.AppliedSpec.Version)
+
+		if err != nil {
+			prometheusErr = err
```

Comment:
I think that it would be nice to add more context to this error, e.g.
```
prometheusErr = pkgerrors.Wrapf(err, "could not reconcile Prometheus, because of unknown SMCP version")
```
WDYT?